CMakeLists.txt

```diff
@@ -150,19 +160,28 @@ if (PREFER_EXTERNAL_DEPS)
     message(WARNING "No ankerl/unordered_dense found in system libraries, using vendored ankerl/unordered_dense...")
   endif()
   # cxxopts
-  if (ENABLE_DATA_TOOLS OR ENABLE_TOOLS)
-    find_package(cxxopts QUIET)
-    if (cxxopts_FOUND)
-      get_target_property(cxxopts_include_dir cxxopts::cxxopts INTERFACE_INCLUDE_DIRECTORIES)
-    else()
-      message(WARNING "No cxxopts found in system libraries, using vendored cxxopts...")
-    endif()
+  find_package(cxxopts QUIET)
+  if (cxxopts_FOUND)
+    get_target_property(cxxopts_include_dir cxxopts::cxxopts INTERFACE_INCLUDE_DIRECTORIES)
+  else()
+    message(WARNING "No cxxopts found in system libraries, using vendored cxxopts...")
+  endif()
+  # libosmium/protozero
+  find_path(OSMIUM_INCLUDE_DIRS "osmium/version.hpp")
+  find_path(PROTOZERO_INCLUDE_DIRS "protozero/version.hpp")
+  if (NOT OSMIUM_INCLUDE_DIRS OR NOT PROTOZERO_INCLUDE_DIRS)
```

Comment:
protozero is a transitive dependency of libosmium. either both of them are in the system libs or we take the vendored ones

Reply:
I ment a bit different thing - `set(libosmium_include_dir ${OSMIUM_INCLUDE_DIRS} ${PROTOZERO_INCLUDE_DIRS})` so they will always be together.

And this approach is inline with the concept of cmake targets - each target should bring own dependencies and client code/cmake should be able to use `libosmium` without caring what exactly it depends on.

Reply:
Ah yes, sorry, now I understand. Sure, that make sense, done in https://github.com/valhalla/valhalla/pull/5455/commits/e179e3a09ca44dc0938102b85c5f835b8ebd7d17